### PR TITLE
Convert `pC` commands to rzshell

### PR DESCRIPTION
**Pull request by XVilka**
**Your checklist for this pull request**
- [x] I've read the [guidelines for contributing](https://github.com/rizinorg/rizin/blob/master/DEVELOPERS.md) to this repository
- [x] I made sure to follow the project's [coding style](https://github.com/rizinorg/rizin/blob/master/DEVELOPERS.md#code-style)
- [x] I've documented or updated the documentation of every function and struct this PR changes. If not so I've explained why.
- [ ] I've added tests that prove my fix is effective or that my feature works (if possible)
- [ ] I've updated the [rizin book](https://github.com/rizinorg/book) with the relevant information (if needed)

**Detailed description**

- Converted `pC` commands to rzshell
- Removed `pCc` command because there is no use for printing C-style array in columns

**Test plan**

CI is green

**Closing issues**

Partially addresses https://github.com/rizinorg/rizin/issues/1590

### CODE REVIEW

File: librz/core/cmd/cmd_print.c [hunk 1]
```diff
@@ -6489,3 +6315,170 @@ RZ_IPI RzCmdStatus rz_print_minus_table_handler(RzCore *core, int argc, const ch
...
+		rz_config_set(core->config, "asm.offset", o_ao);
+		rz_config_set(core->config, "asm.bytes", o_ab);
+	}
+	rz_config_set(core->config, "asm.bytes", o_ab);
+	free(o_ao);
+	free(o_ab);
+	return RZ_CMD_STATUS_OK;
+}
+
+RZ_IPI RzCmdStatus rz_print_columns_Disassembly_handler(RzCore *core, int argc, const char **argv) {
```
Comment (ret2libc) [inline librz/core/cmd/cmd_print.c]: pCD 20
ERROR: Failed to read registers.
ERROR: No such register or register type: ""
ERROR: core: Error while parsing command: `(null) 128@r:SP;pd@ 0x100003a3c`

Comment (ret2libc) [inline librz/core/cmd/cmd_print.c]: also, this and the previous function are so similar that we should probably merge them IMHO.

Comment (ret2libc) [reply librz/core/cmd/cmd_print.c]: and also the next one is somehow similar...

File: librz/core/cmd/cmd_print.c [hunk 2]
```diff
@@ -6489,3 +6315,170 @@ RZ_IPI RzCmdStatus rz_print_minus_table_handler(RzCore *core, int argc, const ch
...
+	rz_config_set_i(core->config, "hex.cols", hex_cols);
+	return true;
+}
+
+RZ_IPI RzCmdStatus rz_print_columns_hex_annotated_handler(RzCore *core, int argc, const char **argv) {
+	int user_rows = argc > 1 ? rz_num_math(core->num, argv[1]) : -1;
+	return bool2status(print_hexdump_columns(core, user_rows, "pxa"));
+}
+
+RZ_IPI RzCmdStatus rz_print_columns_hex_op_colored_handler(RzCore *core, int argc, const char **argv) {
```
Comment (ret2libc) [inline librz/core/cmd/cmd_print.c:6467]: pCA 10
ERROR: core: Invalid length
ERROR: core: Invalid length
ERROR: core: Invalid length

**Review by ret2libc — COMMENTED** [review]
Any super simple test we can add for these commands? You can use scr.columns/scr.rows to force the size of the RzCons so it is reproducible.

File: librz/core/cmd/cmd_print.c (hunk 1 above)
Comment (XVilka) [reply librz/core/cmd/cmd_print.c]: This command works only in debug mode, I limited it to return the error message if not.

Comment (XVilka) [reply librz/core/cmd/cmd_print.c]: They are not too similar - one returns disassembly, while other - debug context.

**Comment by XVilka** [comment]
@ret2libc everything should be fixed now - the problem with `pCA` was that it was using `pxA` which changed behavior since - it limits itself to the block size in the short form (`pxA`) but works fine in the long form (`pxAl`).

**Comment by ret2libc** [comment]
image

The problem is that the second columns starts at 0x100003a64, but the first one ends with 0x100003a5c. So there is an instruction (0x100003a60) that is not displayed.

However... this is not strictly a problem with your conversion, so if you want go ahead but let's create an issue for this at least.

**Comment by XVilka** [comment]
OK: https://github.com/rizinorg/rizin/issues/3318

**Pull request merged**